### PR TITLE
Add webcron to scheduler config

### PR DESCRIPTION
**Pull request by ditsuke**
### Summary of Changes
Auto-generate the webcron key and enhance scheduler config form.

### Testing Instructions
Test the scheduler component configuration for the webcron key. It should auto-generate on enabling webcron and saving the form. Toggling the reset radio should reset the key on form save.
Webcron base link should be copy-able through the link field in the form, which is visible when the webcron feature is enabled and a key has been generated.

### CODE REVIEW

File: plugins/system/schedulerunner/schedulerunner.php [hunk 1]
```diff
@@ -183,15 +218,89 @@ public function runTestCron(Event $event)
...
+		/** @var Extension $table */
+		[$context, $table] = $event->getArguments();
+
+		if ($context !== 'com_config.component'
+			|| $this->app->input->get('component') !== 'com_scheduler')
+		{
+			return;
+		}
+
+		$params = new Registry(json_decode($table->params));
```
Comment (bembelimen) [inline plugins/system/schedulerunner/schedulerunner.php]: I think, ```json_decode``` is not needed.

File: plugins/system/schedulerunner/schedulerunner.php [hunk 2]
```diff
@@ -183,15 +218,89 @@ public function runTestCron(Event $event)
...
 	 */
-	protected function runScheduler(int $id = 0, bool $unpublish = false): ?Task
+	public function generateWebcronKey(EventInterface $event): void
 	{
-		return (new Scheduler)->runTask($id, $unpublish);
+		/** @var Extension $table */
+		[$context, $table] = $event->getArguments();
+
+		if ($context !== 'com_config.component'
+			|| $this->app->input->get('component') !== 'com_scheduler')
```
Comment (bembelimen) [inline plugins/system/schedulerunner/schedulerunner.php]: Did you check, if in ```$table``` the ```component``` is set? Then you could check for that.

File: plugins/system/schedulerunner/schedulerunner.php (hunk 1 above)
Comment (ditsuke) [reply plugins/system/schedulerunner/schedulerunner.php]: Thanks, it was not!

File: plugins/system/schedulerunner/schedulerunner.php (hunk 2 above)
Comment (ditsuke) [reply plugins/system/schedulerunner/schedulerunner.php]: Removed the input check and checking only for the `$table->name`.

**Pull request merged**